client/src/components/Ticketing/ticketingmanager/showings/ShowingUpdated/ShowingUtils.tsx

```diff
@@ -0,0 +1,227 @@
+) => {
+  return async (event, actions) => {
+    actions.setStatus('Submitting...');
+    try {
+      const submitRes = await fetch(url, {
+        credentials: 'include',
+        method: method,
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${token}`,
+        },
+        body: JSON.stringify(event),
+      });
+      actions.setSubmitting(false);
+      if (!submitRes.ok) {
+        throw submitRes;
+      }
+      if (onSuccess) {
+        await onSuccess(submitRes);
+      }
+    } catch (error) {
+      actions.setSubmitting(false);
+      if (onError) {
+        await onError(error);
+      }
+    }
+  };
+};
+// eslint-disable-next-line max-len
+export const createDeleteFunction = (
+  method: string,
+  url: string,
+  token: string,
+  onSuccess?,
+  onError?,
+) => {
+  return async (setIsDeleting) => {
+    try {
+      const deleteRes = await fetch(url, {
+        credentials: 'include',
+        method: method,
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${token}`,
+        },
+      });
+      if (!deleteRes.ok) {
+        throw deleteRes;
+      }
+      setIsDeleting(false);
+      if (onSuccess) {
+        await onSuccess();
+      }
+    } catch (error) {
+      console.log(error);
+      setIsDeleting(false);
+      if (onError) {
+        await onError(error);
+      }
+    }
+  };
+};
+
+export const fetchTicketTypes = async (setTicketTypes) => {
+  try {
+    const ticketTypeRes = await fetch(
+      process.env.REACT_APP_API_1_URL + '/tickets/allTypes',
+    );
+    if (!ticketTypeRes.ok) {
+      throw new Error('Unable to fetch ticket types');
+    }
+    setTicketTypes((await ticketTypeRes.json()).data);
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+export const getEventData = async (eventID, setEventData) => {
+  const eventData = {
+    seasonid_fk: undefined,
+    eventid: undefined,
+    eventname: '',
+    eventdescription: '',
+    active: false,
+    imageurl: '',
+  };
+  await fetch(process.env.REACT_APP_API_1_URL + '/events/' + eventID)
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error('Unable to fetch event');
+      }
+      return response.json();
+    })
+    .then((data) => {
+      eventData.eventname = data.data[0].title;
+      eventData.eventdescription = data.data[0].description;
+      eventData.active = data.data[0].active;
+      eventData.imageurl = data.data[0].imageurl;
+      eventData.eventid = data.data[0].eventid;
+      eventData.seasonid_fk = data.data[0].seasonid_fk;
+    });
+  setEventData(eventData);
+};
+
+export const useFetchEventData = (eventID: number) => {
+  const [eventData, setEventData] = useState(undefined);
+  const [loading, setLoading] = useState(true);
+  const [ticketTypes, setTicketTypes] = useState([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    void fetchTicketTypes(setTicketTypes);
```

Comment:
Is this a typo? Is this a function call?

Reply:
Adding the void keyword just gets rid of the type error thrown because the promise isn't awaited.

Reply:
Oh okay I see, yeah the function just sets the state and doesnt return anything so it defaults to a void promise since its async. That makes sense, thanks for clarifying.